logging.go

```diff
@@ -27,3 +36,93 @@ It can also be used with for example a Logrus logger:
 
 */
 type LogFunc func(format string, args ...interface{})
+
+// stringifyForLog will stringify a amqp.Return, amqp.Delivery or a Request.
+func stringifyForLog(item interface{}) string {
```

Comment:
I don't really like this, feels hackish and with little to no benefit.

Just define multiple methods and you've already solved the signature (no pointers), the sorting etc.

```go
func deliveryAsString(d *amqp.Delivery) string {
    if d == nil {
        return "<nil>"
    }

    parts := []string{
        fmt.Sprintf("%s=%s", "AppID", d.AppId),
        ...,
    }

    return fmt.Sprintf("[%s]", strings.Join(parts, ", "))
}
```

But over all, this kind of operation seems like it's suitable for an application and not a library. I know we discussed logging ages ago but this seems to be more than just logging. What are you even suppose to do with these logs? Index them in Elasticsearch?

Do we need to provide a way to add an event channel? A middleware?

Reply:
To be clear, I'm fine merging this but I want to raise the questions above regarding the way it's implemented and if it's even suitable here.

Reply:
Yea, I think you are right.

I want to log more information about the request on error logs. Then it would be up to the user of this library on how to store them. Right now we only log the correlation id, but that id is only set internally and there is no other way of knowing what request that actually was.